controller/scrape.go

```diff
@@ -40,66 +41,104 @@ var numericStatus = map[string]float64{
+	var total, passed, failed, awaited float64
+	passed = 0
+	failed = 0
+	awaited = 0
+	expStatusList := chaosEngine.Status.Experiments
+	total = float64(len(expStatusList))
+	for i, v := range expStatusList {
+		verdictFloat := getValueFromVerdict(strings.ToLower(v.Verdict))
+		if verdictFloat == 4 {
+			awaited++
+		} else if verdictFloat == 3 {
+			passed++
+		} else if verdictFloat == 2 {
+			failed++
+		} else if verdictFloat == 1 {
+			defineRunningExperimentMetric(chaosEngine.Name, chaosEngine.Namespace, chaosEngine.Spec.Experiments[i].Name)
+		}
+	}
+	return total, passed, failed, awaited
 
-	chaosResult, StatusMap := calculateChaosResult(chaosResultMap)
-	fmt.Printf("%+v\n", StatusMap)
-	totalExpCount := float64(len(engine.Spec.Experiments))
+}
+func defineRunningExperimentMetric(engineName string, engineNamespace string, experimentName string) {
+	klog.V(2).Infof("Running Experiment Metrics: EnginaName: %v, EngineNamespace: %v, ExperimentName: %v", engineName, engineNamespace, experimentName)
+	RunningExperiment.WithLabelValues(engineNamespace, engineName, experimentName, fmt.Sprintf("%s-%s", engineName, experimentName)).Set(float64(1))
 
-	return totalExpCount, chaosResult.TotalPassedExp, chaosResult.TotalFailedExp, StatusMap, nil
 }
 
-//setChaosResultValue will populate the default value of chaos result
-func setChaosResultValue(clientSet *clientV1alpha1.Clientset, chaosResultSpec ChaosResultSpec) map[string]string {
-	chaosResultMap := make(map[string]string)
-	for _, test := range chaosResultSpec.ChaosExperimentList {
-		chaosResultName := fmt.Sprintf("%s-%s", chaosResultSpec.ExporterSpec.ChaosEngine, test)
-		testResultDump, err := clientSet.LitmuschaosV1alpha1().ChaosResults(chaosResultSpec.ExporterSpec.AppNS).Get(chaosResultName, metav1.GetOptions{})
-		if err != nil {
-			if strings.Contains(err.Error(), "not found") {
-				// lack of result cr indicates experiment not executed
-				chaosResultMap[test] = "not-executed"
-			}
-			continue
-		}
-		chaosResultMap[test] = testResultDump.Status.ExperimentStatus.Verdict
+func getValueFromVerdict(verdict string) float64 {
+
+	fmt.Printf("Verdict of the Experiment: %v", verdict)
+	if verdict == "pass" {
+		return 3
+	} else if verdict == "fail" {
+		return 2
+	} else if verdict == "awaited" {
+		return 1
+	} else if verdict == "waiting" {
+		return 4
+	} else {
+		return 0
 	}
-	return chaosResultMap
 }
 
-// calculateChaosResult will calculate the number of pass and failed experiments
-func calculateChaosResult(chaosResult map[string]string) (ChaosExpResult, map[string]float64) {
-	var cr ChaosExpResult
-	StatusMap := make(map[string]float64)
-	for index, status := range chaosResult {
-		if status == StatusPassed {
-			cr.TotalPassedExp++
-		} else if status == StatusFailed {
-			cr.TotalFailedExp++
+func filterMonitoringEnabledEngines(engineList *litmuschaosv1alpha1.ChaosEngineList) *litmuschaosv1alpha1.ChaosEngineList {
```

Comment:
Can we filter while listing using ListOption{}

Reply:
No @chandankumar4 , tried it not possible with CR's

Reply:
okay!!